src/dmtcp_launch.cpp

```diff
@@ -591,6 +593,28 @@ int main ( int argc, char** argv )
   return -1;
 }
 
+static void* start_fnc(void *arg)
+{
+  pthread_exit(0);
+}
+
+extern "C" int __clone(int (*fn) (void *arg), void *child_stack, int flags,
+                       void *arg, int *ptid,
+                       struct user_desc *tls, int *ctid)
+{
+  intptr_t tidOffset = (char*)ptid - (char*)tls;
+  char value[100] = {0};
+  snprintf(value, 100, "%d", tidOffset);
+  setenv(ENV_VAR_TID_OFFSET, value, 1);
+}
+
```

Comment:
@rohgarg shouldn't this overwrite of __clone be calling actual __clone definition ?